### PR TITLE
feat: comment feature for listings (redone to remove unnecessary changes😃)

### PR DESCRIPTION
**Pull request by VictorCodebase**
I have successfully managed to get back the version before the one with 185 files changed.

I have again really tried to see why 296 of listings.json is being flagged as changed. They are pretty much similar.
I even compared kinda literally:
image

The code still runs as expected:
image

### CODE REVIEW

File: packages/portal/src/app/rentalpage/comment-generator.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    return comment;
+  } else {
+    const comment = negative[Math.floor(Math.random() * (negative.length - 1))];
+    return comment;
+  }
+}
+
+export function generateCommentor(): string {
+  const user = users[Math.floor(Math.random() * (users.length - 1))];
+  return user;
```
Comment (manekinekko) [inline packages/portal/src/app/rentalpage/comment-generator.ts:58]:
```suggestion
  return users[Math.floor(Math.random() * (users.length - 1))];
```

File: packages/portal/src/app/rentalpage/comment-generator.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+export function generateTime(): string {
+  const randomTime = time[Math.floor(Math.random() * (time.length - 1))];
+  return randomTime;
+}
+
+export function randomLikeDislike(max: number): number {
+  {
+    const random = Math.floor(Math.random() * max);
+    return random;
```
Comment (manekinekko) [inline packages/portal/src/app/rentalpage/comment-generator.ts:69]:
```suggestion
    return Math.floor(Math.random() * max);
```

File: packages/portal/src/app/rentalpage/comment-generator.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+}
+
+export function generateCommentor(): string {
+  const user = users[Math.floor(Math.random() * (users.length - 1))];
+  return user;
+}
+
+export function generateTime(): string {
+  const randomTime = time[Math.floor(Math.random() * (time.length - 1))];
+  return randomTime;
```
Comment (manekinekko) [inline packages/portal/src/app/rentalpage/comment-generator.ts:63]:
```suggestion
  return time[Math.floor(Math.random() * (time.length - 1))];
```

File: packages/portal/src/app/rentalpage/comment-generator.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  "1/03/2020",
+];
+
+export function generateComments(starRating: number): string {
+  if (starRating >= 3) {
+    const comment = positive[Math.floor(Math.random() * (positive.length - 1))];
+    return comment;
+  } else {
+    const comment = negative[Math.floor(Math.random() * (negative.length - 1))];
+    return comment;
```
Comment (manekinekko) [inline packages/portal/src/app/rentalpage/comment-generator.ts:52]:
```suggestion
    return negative[Math.floor(Math.random() * (negative.length - 1))];
```

File: packages/portal/src/app/rentalpage/comment-generator.ts [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  "1/11/2021",
+  "10/01/2021",
+  "08/04/2020",
+  "1/03/2020",
+];
+
+export function generateComments(starRating: number): string {
+  if (starRating >= 3) {
+    const comment = positive[Math.floor(Math.random() * (positive.length - 1))];
+    return comment;
```
Comment (manekinekko) [inline packages/portal/src/app/rentalpage/comment-generator.ts:49]:
```suggestion
    return positive[Math.floor(Math.random() * (positive.length - 1))];
```

File: packages/portal/src/app/rentalpage/rentalpage.component.ts [hunk 6]
```diff
@@ -24,6 +25,11 @@ export class RentalpageComponent implements OnInit {
 
   listing = signal<Listing>({} as Listing);
   reviewStars = signal<number[]>([]);
+  comments: string[] = [];
+  commentors: string[] = [];
+  commentTime: string[] = [];
+  likes: number[] = [];
+  dislikes: number[] = [];
```
Comment (manekinekko) [inline packages/portal/src/app/rentalpage/rentalpage.component.ts:32]: can all of these be [Angular signals](https://angular.io/guide/signals)?

**Review by manekinekko — CHANGES_REQUESTED** [review]

**Comment by VictorCodebase** [comment]
Thanks so much @manekinekko for pointing out area I could improve. I really appreciate it thanks!

Ive incorporated the changes

File: packages/portal/src/app/rentalpage/rentalpage.component.ts [hunk 7]
```diff
@@ -24,9 +25,15 @@ export class RentalpageComponent implements OnInit {
...
   reviewStars = signal<number[]>([]);
+  comments = signal<string[]>([]);
+  commentors = signal<string[]>([]);
+  commentTime = signal<string[]>([]);
+  likes = signal<number[]>([]);
+  dislikes = signal<number[]>([]);
   isLoading = signal(true);
 
   private router = inject(Router);
+  private route = inject(ActivatedRoute);
```
Comment (manekinekko) [inline packages/portal/src/app/rentalpage/rentalpage.component.ts:36]: `route` is not used. You can remove it.

File: packages/portal/src/app/rentalpage/rentalpage.component.html [hunk 8]
```diff
@@ -57,31 +84,39 @@
...
+            <!-- {{ comment.user }} -->
+            <span>{{ commentors()[i] }}</span>
+          </div>
+          <div class="listing-comments__comment__header__date">
+            <!-- <span>{{ comment.date }}</span> -->
+            <span>{{ commentTime()[i] }}</span>
+          </div>
+        </div>
+        <div class="listing-comments__comment__body">
+          <span>{{ comment }}</span>
```
Comment (manekinekko) [inline packages/portal/src/app/rentalpage/rentalpage.component.html:103]:
```suggestion
          <span class="listing-comments__comment__body__text">{{ comment }}</span>
```

File: packages/portal/src/app/rentalpage/rentalpage.component.scss [hunk 9]
```diff
@@ -183,3 +185,118 @@
...
+    }
+
+    .listing-comments__comment__body {
+      display: flex;
+      flex-direction: row;
+      align-items: center;
+      justify-content: left;
+
+      .listing-comments__comment__body__text {
+        color: #999;
```
Comment (manekinekko) [inline packages/portal/src/app/rentalpage/rentalpage.component.scss:241]:
```suggestion
        color: #000;
```

File: packages/portal/src/app/rentalpage/comment-generator.ts [hunk 10]
```diff
@@ -0,0 +1,66 @@
+//import a way to get random numbers
```
Comment (manekinekko) [inline packages/portal/src/app/rentalpage/comment-generator.ts:1]: can you convert this file into an [Angular service](https://angular.io/guide/architecture-services)?

File: packages/portal/src/app/rentalpage/rentalpage.component.ts [hunk 11]
```diff
@@ -54,9 +61,44 @@ export class RentalpageComponent implements OnInit {
...
+    //   .map(() => generateTime()));
+    // this.likes.set(Array(limit)
+    //   .fill(0)
+    //   .map(() => randomLikeDislike(100)));
+    // this.dislikes.set(Array(limit)
+    //   .fill(0)
+    //   .map(() => randomLikeDislike(40)));
+
+    //! ive commented this block out because I felt it might not be really concincise or easily readable
+    //! I can revert it back if its better that way
```
Comment (manekinekko) [inline packages/portal/src/app/rentalpage/rentalpage.component.ts:89]: you can remove this comment. The code below is more maintainable.

```suggestion
```

File: packages/portal/src/app/rentalpage/rentalpage.component.scss [hunk 12]
```diff
@@ -183,3 +185,118 @@
...
+  flex-direction: column;
+  align-items: flex-start;
+  justify-content: left;
+  padding: 10px 0;
+  border: solid 1px #ccc;
+  border-radius: 0 0 13px 13px;
+  min-height: fit-content;
+  max-height: 400px;
+  overflow-y: scroll;
+
```
Comment (manekinekko) [inline packages/portal/src/app/rentalpage/rentalpage.component.scss:201]:
```suggestion
  & > div {
    width: 100%;
  }
```

File: packages/portal/src/app/rentalpage/rentalpage.component.scss [hunk 13]
```diff
@@ -183,3 +185,118 @@
...
+  .listing-comments__comment {
+    display: flex;
+    flex-direction: column;
+    align-items: flex-start;
+    justify-content: left;
+    padding: 10px 0;
+    border-bottom: solid 1px #ccc;
+    width: 730px;
```
Comment (manekinekko) [inline packages/portal/src/app/rentalpage/rentalpage.component.scss:209]:
```suggestion
```

**Review by manekinekko — CHANGES_REQUESTED** [review]

**Comment by VictorCodebase** [comment]
Im so excited to say that Ive managed to make all improvements suggested - including the changing comment-generator file to Angular service one.

**Comment by VictorCodebase** [comment]
Thank you so much @anfibiacreativa , @manekinekko and @sinedied  for taking us through the hackathon. I really learnt a lot to do with collaboration, using GitHub, Copilot, using Azure, using JS frameworks like azure (I never knew what to learn) and like so so much more.

I couldn't be more grateful😀

**Pull request closed without merging**